Fix pack module (#169)
tarantool/tt: doc/examples.rst

    require("fiber").sleep(1)

Packing environments
----------------------------------

> For example, we want to pack a single application. If there is the next environment structure::

Reviewer: ```suggestion
For example, we want to pack a single application. Here is the content of the sample application::
```

Author: Fixed, thanks.

      single_environment/
      ├── tarantool.yaml
      └── init.lua

> The contents of ``./tarantool.yaml`` is supposed to be:

Reviewer: "The content"
Or just use a file name:
``tarantool.yaml``:
Also, I think it is better to omit ./ before file name.

Author: Fixed.

.. code-block:: yaml

    tt:
        app:

For packing it into tarball, call:

.. code-block:: bash

   $ tt pack tgz
      • Apps to pack: single_environment
      • Generating new tarantool.yaml for the new package

>       • The passed bundle packed into temporary directory: /var/folders/T/tt_pack1901831662

Reviewer: You have changed log level of this message. Looks like, it should not be printed bye default. Please, update examples.

Author: Fixed, thank you.

      │   └── modules
      ├── instances_enabled
      │   └── single_environment -> ../single_environment
      └── var
          ├── lib
          ├── log
          └── run

Example of packing a multi-app environment::

     bundle/
     ├── tarantool.yaml
     ├── env
     │   ├── bin
     │   │   ├── tt
     │   │   └── tarantool
     │   └── modules
     ├── myapp
     │   ├── Dockerfile.build.cartridge
     │   ├── Dockerfile.cartridge
     │   ├── README.md
     │   ├── app
     │   ├── bin
     │   ├── deps.sh
     │   ├── failover.yml
     │   ├── init.lua
     │   ├── instances.yml
     │   ├── myapp-scm-1.rockspec
     │   ├── pack-cache-config.yml
     │   ├── package-deps.txt
     │   ├── replicasets.yml
     │   ├── stateboard.init.lua
     │   ├── systemd-unit-params.yml
     │   ├── tarantool.yaml
     │   ├── test
     │   └── tmp
     ├── myapp2
     │   ├── app.lua
     │   ├── data
     │   ├── etc
     │   ├── myapp2
     │   ├── queue
     │   ├── queue1.lua
     │   └── queue2.lua
     ├── myapp3.lua
     ├── app4.lua
     ├── instances_enabled
     │   ├── app1 -> ../myapp
     │   ├── app2 -> ../myapp2
     │   ├── app3.lua -> ../myapp3.lua

>      │   ├── app4.lua -> /Users/dev/tt_demo/bundle1/app4.lua

Reviewer: Is this tree an example of packed bundle? How is it possible to have an absolute path for symlink here?

     │   └── app5.lua -> ../myapp3.lua
     └── var
         ├── lib
         ├── log

>          ├── pid

Reviewer: Why is pid dir missing in previous "single app" example?

Author: Pid is an artifact here, removed.

``tarantool.yaml``:

.. code-block:: yaml

    tt:

>         app:

Reviewer: Suppose, this will not work. Because instances_enabled is . by default, which is single-app in current dir. But we have multi-app bundle here.

Author: Fixed, thank you.

> Absolute symlinks to target files/directories will be copied to the package.

Reviewer: Why do we copy absolute symlinks to the package?

Author: Sorry, I have copy pasted your rephrasing sentence, now fixed. We resolve any absolute symlink and replace it to the relative one.

> Pay attention, that all full links from instances enabled will be resolved, all sources will be copied

Reviewer: Pls, use "absolute symlink/symbolic link" instead of "full link".
And I'd like to re-phrase this to something like "Absolute symlinks target files/directories will be copied to the package. All relative symlinks in inst_enabled and in apps directories will be converted relatively to the package root directory."

Author: Fixed.

and in the result package, instances_enabled directory will contain only relative links.

For packing deb package call:

.. code-block:: bash

   $ tt pack deb --name dev_bundle --version 1.0.0
   • A root for package is located in: /var/folders/c6/jv1r5h211dn1280d75pmdqy80000gp/T/2166098848
      • Apps to pack: app1 app2 app3 app4 app5

   myapp scm-1 is now installed in /var/folders/c6/jv1r5h211dn1280d75pmdqy80000gp/T/tt_pack4173588242/myapp/.rocks

      • myapp rocks are built successfully
      • Generating new tarantool.yaml for the new package
      • Initialize the app directory for prefix: data/usr/share/tarantool/bundle
      • Create data tgz
      • Created control in /var/folders/***/control_dir
      • Created result DEB package: /var/folders/***/T/tt_pack4173588242

> Now the result package may be distributed and unpacked by dpkg command.

Reviewer: Unpacked by -> installed using?

Author: Fixed.
